android/MobileSdk/src/main/java/com/spruceid/mobile/sdk/nfc/BaseNfcPresentationService.kt

```diff
@@ -0,0 +1,234 @@
+        private var inNegotiation = false
+        private var _apduHandoverDriver: ApduHandoverDriver? = null
+        private val apduHandoverDriver: ApduHandoverDriver
+            get() {
+                if (_apduHandoverDriver == null) {
+                    _apduHandoverDriver =
+                        // Use static handover, temporarily
+                        ApduHandoverDriver(negotiated = false, strict = false)
+                }
+                return _apduHandoverDriver!!
+            }
+    }
+
+    private fun defer(delay: Duration, action: Runnable) {
+        Handler(Looper.getMainLooper()).postDelayed(action, delay.inWholeMilliseconds)
+    }
+
+    override fun processCommandApdu(commandApdu: ByteArray, extras: Bundle?): ByteArray? {
+
+        if(!shouldPerformHandoverEngagement()) return null;
+
+        if (resetQueued) {
+            resetQueued = false
+            apduHandoverDriver.reset()
+        }
+
+        val isReadBinaryCommand = commandApdu.size > 4 && commandApdu[1] == 0xB0.toByte()
+
+        currentInteractionId++
+
+        // Read binary commands shouldn't affect state flags,
+        // since they may come after the success state is reached.
+        if (!isReadBinaryCommand) {
+            if (!inNegotiation) {
+                negotiationStarted()
+                inNegotiation = true
+            }
+            
+            doNotNotifyOnDisconnect = false
+            negotiationFailedFlag = false
+        }
+
+        NfcListenManager.setExpectedNdefFromHandover(true, applicationContext, componentName())
+
+        val ret = apduHandoverDriver.processApdu(commandApdu)
+        val carrierInfo = apduHandoverDriver.getCarrierInfo()
+        if (carrierInfo != null) {
+            Handler(Looper.getMainLooper()).post { negotiatedTransport(carrierInfo) }
+            doNotNotifyOnDisconnect = true
+        }
+        val returnedApduSuccessStatus =
+            ret.size >= 2 &&
+                    ret[ret.size - 2] == (0x90.toByte()) &&
+                    ret[ret.size - 1] == (0x00.toByte())
+
+        // If negotiation failed, send the negotiation failed message to the user
+        // and flag that an error message has already been sent.
+        // The error flag prevents a double error notification upon NFC disconnect.
+        // We defer the actual "Negotiation Failed" message because some readers try multiple
+        // handover techniques before giving up.
+        if (!returnedApduSuccessStatus) {
+            doNotNotifyOnDisconnect = true
+
+            // We don't want to report errors returned for SELECT AID to the user
+            // because this probably means that both:
+            //  1. the APDU handover driver is in strict mode
+            //  2. the device communicating with us over NFC is not an mDOC reader
+            val isSelectAidCommand = commandApdu.size < 4 && commandApdu[0] == 0xA4.toByte() && commandApdu[2] == 0x04.toByte()
```

Comment:
Could you make those hardcoded bytes (`0xB0`, `0x90`, `0x00`, `0xA4`, `0x04`) consts to make it easier to debug later on? The code is very self explanatory thanks to the variable names, it's just in this case I was suprised `commandApdu[1]` was skipped